src/compiler/crystal/semantic/bindings.cr

```diff
@@ -215,6 +224,22 @@ module Crystal
       type
     end
 
+    # Computes the type resulting from assigning type to freeze_type,
+    # in the case where freeze_type is not nil.
+    #
+    # Special cases are listed inside the method body.
+    def assign_type(freeze_type, type)
```

Comment:
I'd suggest renaming this method due to the fact it doesn't actually _do_ anything, it just returns the desired type:

```suggestion
    def assignment_type(freeze_type, type)
```

Reply:
Thanks! Yeah, I'll actually rename it to something more descriptive.

Reply:
On second thought, I prefer to leave it like that. We have an `Crystal::Assign` ASTNode that represent an assignment. It's called Assign... I don't know why. But I prefer to leave that name too for the method so it's clear.

Also, we have `map_type` a few lines above. It maps the type to another type. This just assigns the type to `freeze_type` and returns the resulting type. I think it's clear (from a compiler perspective).

Reply:
Well, I ended up renaming it 🤷‍♂